### PR TITLE
Issue 92 weird app behavior

### LINKED ISSUE #92: Weird app behaviour
# Description
The app is behaving very weirdly in different situations. One that I encountered -
1. Add a book to the cart (notice how the cart icon disappears)
2. Go to the cart screen (notice how it loads for a long time) and place an order/delete the items
3. Go back and open any books (It won't open)
4. The app is stuck in a weird place where nothing is working

The bug is somewhere between the commit 3a3c330a96ebfb5b6242b8cc3cb4a7eae570a623 and the latest commit for sure.

### PR DESCRIPTION
**Pull request by mortal-ghost**
## Program
- [X] CodePeak
- [ ] Script Winter of Code
- [ ] None

## Issue
Fixes #92 

## Description

Most of the issues seemed to be created by the _key.currentState.showSnackBar which is replaced by ScaffoldMessenger.of(context).showSnackBar.
Issue with confirming an order on dialog was because of the same reason as above but also needed the context of the underlying Scaffold rather than the dialog context which is declared as a new variable. 
The issues mentioned are resolved, I did not face any other issues.

## Type of change
You can write manually also

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] The app runs properly
- [X] I have performed a self-review of my own code
- [X] I have formatted and commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
Will review after #95 is merged.

File: pubspec.lock [hunk 1]
```diff
@@ -7,7 +7,7 @@ packages:
       name: async
```
Comment (Saransh-cpp) [inline pubspec.lock:7]: Please revert back the changes in `pubspec.lock`

**Comment by Saransh-cpp** [comment]
@mortal-ghost, please merge the master branch of my repository in this PR

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @mortal-ghost! Awesome work! This was a hard-to-spot bug. A few minor changes -
- There are other `SnackBar`s in `product_details.dart` that need these changes.
- You will need to do the same for the `Favourite` screen (again, just `SnackBar`s and probably the `context`), plus remove the `MaterialApp` parent widget from that screen.

**Comment by mortal-ghost** [comment]
> Thanks, @mortal-ghost! Awesome work! This was a hard-to-spot bug. A few minor changes -
> 
> * There are other `SnackBar`s in `product_details.dart` that need these changes.
> * You will need to do the same for the `Favourite` screen (again, just `SnackBar`s and probably the `context`), plus remove the `MaterialApp` parent widget from that screen.

I have made the changes. On the product_details.dart, since the snackbar is displayed on the underlying Scaffold, Navigator pops the dialog first before displaying the snackbar. This is inline with the cart checkout.

**Review by Saransh-cpp — APPROVED** [review]
Thanks, @mortal-ghost! Looks good!

**Pull request merged**